### PR TITLE
fix: copy cell should generate html & plain

### PR DESCRIPTION
**Pull request by ybzky**
close #xxx

## Pull Request Checklist

- [ ] Related tickets or issues have been linked in the PR description (or missing issue).
- [ ] [Naming convention](https://github.com/dream-num/univer/blob/dev/docs/NAMING_CONVENTION.md) is followed (**do please** check it especially when you created new plugins, commands and resources).
- [ ] Unit tests have been added for the changes (if applicable).
- [ ] Breaking changes have been documented (or no breaking changes introduced in this PR).

### CODE REVIEW

**Review by yuhongz — APPROVED** [review]

File: packages/core/src/sheets/worksheet.ts [hunk 1]
```diff
@@ -534,25 +535,39 @@ export class Worksheet {
...
-                returnCellMatrix.setValue(row, col, v);
+            let cellData: Nullable<ICellDataWithExtraData>;
+            if (dataMode === CellModeEnum.Raw) {
+                cellData = this.getCellRaw(row, col);
+            } else if (dataMode === CellModeEnum.Intercepted) {
+                cellData = this.getCell(row, col);
+            } else if (dataMode === CellModeEnum.Both) {
+                cellData = this.getCellRaw(row, col);
+                const displayV = this.getCell(row, col)?.v;
+                if (displayV && cellData) {
```
Comment (VicKun4937) [inline packages/core/src/sheets/worksheet.ts:564]: v = 0

File: packages/sheets-ui/src/services/clipboard/clipboard.service.ts [hunk 2]
```diff
@@ -1121,7 +1120,10 @@ function getMatrixPlainText(matrix: ObjectMatrix<ICellDataWithSpanInfo>) {
     return plain;
 }
 
-function getCellTextForClipboard(cell: ICellDataWithSpanInfo) {
+function getCellTextForClipboard(cell: ICellDataWithExtraData) {
+    if (cell.displayV) {
```
Comment (VicKun4937) [inline packages/sheets-ui/src/services/clipboard/clipboard.service.ts:1124]: here

**Review by Gggpound — APPROVED** [review]

**Review by VicKun4937 — APPROVED** [review]

File: packages/core/src/sheets/worksheet.ts [hunk 3]
```diff
@@ -534,25 +535,39 @@ export class Worksheet {
      *
      * Notice that `ICellData` here is not after copying. In another word, the object matrix here should be
      * considered as a slice of the original worksheet data matrix.
+     *
+     * Control the v attribute in the return cellData.v through dataMode
      */
     getMatrixWithMergedCells(
```
Comment (wzhudev) [inline packages/core/src/sheets/worksheet.ts:541]: Use function overload here to make the return value more predictable.

File: packages/core/src/sheets/typedef.ts [hunk 4]
```diff
@@ -771,3 +771,11 @@ export function getCellInfoInMergeData(row: number, column: number, mergeData?:
         startColumn: mergeColumn,
     };
 }
+
+export type ICellDataWithExtraData = ICellData & { rowSpan?: number; colSpan?: number; displayV?: string };
```
Comment (wzhudev) [inline packages/core/src/sheets/typedef.ts:775]: This is a bad name. The `Extra` does tell anything meaningful to a reader.

**Review by wzhudev — APPROVED** [review]

**Pull request merged**